skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,320 @@
+            raise ValueError(f"{self.coefficients}. Compute the `estimate`")
+        coeffs = self.coefficients
+        coords = np.array(coords)
+
+        if not coords.ndim == 2 or coords.shape[1] != 2:
+            raise ValueError("Input 'coords' must have shape (N,2)")
+
+        x_warp = self._spline_function(coords[:, 0], coords[:, 1], coeffs[:, 0])
+        y_warp = self._spline_function(coords[:, 0], coords[:, 1], coeffs[:, 1])
+        return np.vstack([x_warp, y_warp]).T
+
+    @property
+    def inverse(self):
+        raise NotImplementedError("This is yet to be implemented.")
+
+    def estimate(self, src, dst):
+        """Estimate optimal coefficients that describes the deformation of the points.
+
+
+        Parameters
+        ----------
+        src : (N, 2) array_like
+            Control point at source coordinates
+        dst : (N, 2) array_like
+            Control point at destination coordinates
+
+        Returns
+        -------
+        success: bool
+            True, indicates that the coefficients were successfully estimated.
+
+        Notes
+        -----
+        -  The number of source and destination points must match (N).
+        """
+
+        src = _ensure_2d(src)
+        dst = _ensure_2d(dst)
+
+        if src.shape != dst.shape:
+            raise ValueError("src and dst shape must be identical")
+
+        self.src = src
+        self.dst = dst
+        n, d = src.shape
+
+        K = self._radial_distance(src)
+        P = np.hstack([np.ones((n, 1)), src])
+        L = np.zeros((n + 3, n + 3), dtype=np.float32)
+        L[:n, :n] = K
+        L[:n, -3:] = P
+        L[-3:, :n] = P.T
+        V = np.concatenate([dst, np.zeros((d + 1, d))])
+        self.coefficients = np.dot(np.linalg.inv(L), V)
+        self._estimated = True
+        return self._estimated
+
+    def _spline_function(self, x, y, coeffs):
+        """Solve the spline function in the X and Y directions"""
+        w = coeffs[:-3]
+        a1, ax, ay = coeffs[-3:]
+        Pi_x = self.src[:, 0]
+        Pi_y = self.src[:, 1]
+
+        dx = Pi_x[:, np.newaxis] - x
+        dy = Pi_y[:, np.newaxis] - y
+        r = np.sqrt(dx**2 + dy**2)
+        radial_dist = _radial_basis_function(r)
+        summation = np.dot(w, radial_dist)
+        return a1 + ax * x + ay * y + summation
+
+    def _radial_distance(self, points):
+        """Compute the pairwise radial distances of the given points to the control points.
+
+        Parameters
+        ----------
+        points : ndarray
+            N points in the source space
+        Returns
+        -------
+        ndarray :
+            The radial distance for each `N` point to a control point.
+        """
+        dist = sp.spatial.distance.cdist(self.src, points)
+        return _radial_basis_function(dist)
+
+
+def _radial_basis_function(r):
+    """Compute basis kernel function for thine-plate splines.
+
+    Parameters
+    ----------
+    r : ndarray
```

Comment:
```suggestion
    r : (4, N) ndarray
```
`N` being the number of coordinates passed to `TpsTransform.__call__`.